lib/src/impl_ffi/impl_ffi.utils.dart

```diff
@@ -14,40 +14,101 @@
 
 part of impl_ffi;
 
-/// Attach a finalizer for [key], this means that [ssl.EVP_PKEY_free] will
-/// automatically be called with [key] is garbage collected by Dart.
-///
-/// This takes ownership of [key], and the caller **may not** call
-/// [ssl.EVP_PKEY_free].
+/// Function that can't be inlined to be used for preventing [obj] from being
+/// garbage collected.
+@pragma('vm:never-inline')
+Object _finalizerReachabilityFence(Object obj) {
+  return obj;
+}
```

Comment:
@dcharkes is this really the way to do things?

Reply:
From Dart 2.17 we should have `implements Finalizable` which will automatically add reachability fences at the end of the syntactic scope for all `Finalizable`s. https://dart-review.googlesource.com/c/sdk/+/227501